controllers/machinehealthcheck_controller_test.go

```diff
@@ -0,0 +1,193 @@
+				}
+				return mhc.GetLabels()
+			}, timeout).Should(Equal(ltc.expected))
+		},
+		Entry("when no existing labels exist", labelTestCase{
+			original: map[string]string{},
+			expected: map[string]string{clusterv1.ClusterLabelName: testCluster.GetName()},
+		}),
+		Entry("when the label has the wrong value", labelTestCase{
+			original: map[string]string{clusterv1.ClusterLabelName: "wrong"},
+			expected: map[string]string{clusterv1.ClusterLabelName: testCluster.GetName()},
+		}),
+		Entry("without modifying other labels", labelTestCase{
+			original: map[string]string{"other": "label"},
+			expected: map[string]string{"other": "label", clusterv1.ClusterLabelName: testCluster.ObjectMeta.Name},
+		}),
+	)
+
+	type ownerReferenceTestCase struct {
+		original []metav1.OwnerReference
+		// Use a function so that runtime information can be populated (eg UID)
+		expected func() []metav1.OwnerReference
+	}
+
+	DescribeTable("should ensure an owner reference is present",
+		func(ortc ownerReferenceTestCase) {
+			By("Creating a MachineHealthCheck")
+			mhcToCreate := &clusterv1.MachineHealthCheck{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            "test-mhc",
+					Namespace:       namespace.ObjectMeta.Name,
+					OwnerReferences: ortc.original,
+				},
+				Spec: clusterv1.MachineHealthCheckSpec{
+					ClusterName: testCluster.ObjectMeta.Name,
+					UnhealthyConditions: []clusterv1.UnhealthyCondition{
+						{
+							Type:    corev1.NodeReady,
+							Status:  corev1.ConditionUnknown,
+							Timeout: metav1.Duration{Duration: 5 * time.Minute},
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, mhcToCreate)).To(Succeed())
+
+			Eventually(func() []metav1.OwnerReference {
+				mhc := &clusterv1.MachineHealthCheck{}
+				err := k8sClient.Get(ctx, client.ObjectKey{Namespace: mhcToCreate.GetNamespace(), Name: mhcToCreate.GetName()}, mhc)
+				if err != nil {
+					return []metav1.OwnerReference{}
+				}
+				return mhc.GetOwnerReferences()
+			}, timeout).Should(ConsistOf(ortc.expected()))
+		},
+		Entry("when no existing owner references exist", ownerReferenceTestCase{
+			original: []metav1.OwnerReference{},
+			expected: func() []metav1.OwnerReference {
+				return []metav1.OwnerReference{ownerReferenceForCluster(ctx, testCluster)}
+			},
+		}),
+		Entry("when modifying existing owner references", ownerReferenceTestCase{
+			original: []metav1.OwnerReference{{Kind: "Foo", APIVersion: "foo.bar.baz/v1", Name: "Bar", UID: "12345"}},
+			expected: func() []metav1.OwnerReference {
+				return []metav1.OwnerReference{{Kind: "Foo", APIVersion: "foo.bar.baz/v1", Name: "Bar", UID: "12345"}, ownerReferenceForCluster(ctx, testCluster)}
+			},
+		}),
+	)
+
+})
+
+func cleanupTestMachineHealthChecks(ctx context.Context, c client.Client) error {
+	mhcList := &clusterv1.MachineHealthCheckList{}
+	err := c.List(ctx, mhcList)
+	if err != nil {
+		return err
+	}
+	for _, mhc := range mhcList.Items {
+		err = c.Delete(ctx, mhc.DeepCopy())
```

Comment:
Why DeepCopy?

Reply:
This one is because I needed a pointer and `mhc` is a range variable, I've declared a second variable in the scope and that works, don't know if that's better or not?